dot/core/service.go

```diff
@@ -520,17 +530,21 @@ func (s *Service) HandleSubmittedExtrinsic(ext types.Extrinsic) error {
 		return err
 	}
 
-	rt.SetContextStorage(ts)
-
 	externalExt, err := s.buildExternalTransaction(rt, ext)
 	if err != nil {
 		return fmt.Errorf("building external transaction: %w", err)
 	}
 
+	// ValidateTransaction modifies the trie state so we want to snapshot it
+	// so that the original trie state remains unaffected.
+	temporaryState := ts.Snapshot()
+	rt.SetContextStorage(temporaryState)
 	transactionValidity, err := rt.ValidateTransaction(externalExt)
 	if err != nil {
 		return err
 	}
+	// TODO adding this fails polkadotjs tests
```

Comment:
could an issue be created for this TODO?

Reply:
Just fishing for a reviewer that may have an answer on this. Otherwise yes I'll create an issue for this.

Reply:
Somehow this got fixed and the storage can now be cleared after this runtime call.